packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,250 @@
+    },
+  }
+
+  logger.debug('Searching events by entity and entity ID. Query params: ', params)
+  const result = await dynamoDB.query(params).promise()
+  return (result.Items as Array<EventEnvelope>) ?? []
+}
+
+interface EventStoreKeys {
+  [eventsStoreAttributes.partitionKey]: string
+  [eventsStoreAttributes.sortKey]: string
+}
+
+async function searchEventsByEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entity: string,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: manage pagination
+  // Fist query the index
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    IndexName: eventsStoreAttributes.indexByEntity.name(config),
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.indexByEntity.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': partitionKeyForIndexByEntity(entity, 'event'),
+    },
+  }
+
+  logger.debug('Searching events by entity. Index query params: ', params)
+  const partialResult = await dynamoDB.query(params).promise()
+  const indexRecords = (partialResult.Items as Array<EventStoreKeys>) ?? []
+  return findEventsDataWithKeys(dynamoDB, config, logger, indexRecords)
+}
+
+async function searchEventsByType(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  type: string,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: manage pagination
+  // Fist query the index
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    IndexName: eventsStoreAttributes.indexByType.name(config),
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.indexByType.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': type,
+    },
+  }
+
+  logger.debug('Searching events by type. Index query params: ', params)
+  const partialResult = await dynamoDB.query(params).promise()
+  const indexRecords = (partialResult.Items as Array<EventStoreKeys>) ?? []
+  return findEventsDataWithKeys(dynamoDB, config, logger, indexRecords)
+}
+
+async function findEventsDataWithKeys(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  keys: Array<EventStoreKeys>
+): Promise<Array<EventEnvelope>> {
+  // TODO: Manage pagination
+  const result: Array<EventEnvelope> = []
+
+  const keysBatches = inChunksOf(dynamoDbBatchGetLimit, keys)
+  logger.debug(`Performing batch get for ${keysBatches.length} batches`)
+  for (const keysBatch of keysBatches) {
+    const batchResult = await performBatchGet(dynamoDB, config, logger, keysBatch)
```

Comment:
The `batchGet` operation in Dynamo only allows you to get 100 events at once, that's why we need to do it in chunks

Reply:
You can actually do it with a `do..while` and not need to count the batches

Reply:
😳 How is so? Could you please elaborate a bit?